src/services/codefixes/convertFunctionToEs6Class.ts

```diff
@@ -0,0 +1,191 @@
+/* @internal */
+namespace ts.codefix {
+    registerCodeFix({
+        errorCodes: [Diagnostics.Convert_function_0_to_ES6_class.code],
+        getCodeActions,
+        createCodeFixDiagnosticIfApplicable
+    });
+
+    function createCodeFixDiagnosticIfApplicable(node: Node, context: CodeFixDiagnoseContext): Diagnostic | undefined {
+        if (!isSourceFileJavaScript(context.boundSourceFile)) {
+            return undefined;
+        }
+
+        const checker = context.program.getTypeChecker();
+        const symbol = checker.getSymbolAtLocation(node);
+        if (isClassLikeSymbol(symbol)) {

```

Comment:
i would say this should be limited to the declaration, not on every reference.

I would also say you do not need the checker here.